python_modules/dagster/dagster/_core/definitions/materialize.py

```diff
@@ -41,6 +44,17 @@ def materialize(
             The string partition key that specifies the run config to execute. Can only be used
             to select run config for assets with partitioned config.
         tags (Optional[Mapping[str, str]]): Tags for the run.
+        selection (Optional[Union[str, Sequence[str], Sequence[AssetKey], Sequence[Union[AssetsDefinition, SourceAsset]], AssetSelection]]):
+            A sub-selection of assets to materialize.
+
+            The selected assets must all be included in the assets that are passed to the assets
+            argument.
+
+            If not provided, then all assets will be materialized.
+
+            The string "my_asset*" selects my_asset and all downstream assets within the code
+            location. A list of strings represents the union of all assets selected by strings
+            within the list.
```

Comment:
This is user-facing and IIUC solves the annoying problem of needing to provide source asset versions of assets you don't want to materialize in `assets`. Think it would benefit from an example to be crystal clear.

Reply:
> Seems odd to reference just this one part of the selection syntax. Can we provide a link to a comprehensive description of it?

This is copy/pasted from the selection argument of `define_asset_job`. But yeah this is kind of weird.

I pushed a change that adds an example and fixes up the docs.